### PR TITLE
incooperate nCtrl and webui

### PR DESCRIPTION
**Pull request by rroohhh**
This pull request adds a new control daemon [nctrl](https://github.com/apertus-open-source-cinema/nctrl) and a [web interface](https://github.com/apertus-open-source-cinema/webui) that uses it.

### control daemon
The new control daemon provides a centralized abstraction layer for all the hardware peripherals, that allows for 
- easier readable programs / scripts that deal with the peripherals, as every register has a name and most special values also are defined by name, so less "magic" constants
- exclusive access to devices, to prevent interference from concurrent access to the peripheral
- better off camera testing, as all peripherals support a "mock" mode where no actual reads / writes a executed, but dummy data is presented. By extending this feature it should be possible to provide high level emulation of several important peripherals that could aid automatic testing of software using these peripherals and support developers without access to camera hardware.

While the control daemon will be the future way to access all hardware peripherals, all the old scripts / programs are still usable, but should be ported to the control daemon. In addition to that we should consider supporting the legacy commands like `fil_reg`, `hdmi_reg` and so on, to allow users to transition to the new system more gracefully.

To learn more about the control daemon please refer to its [readme](https://github.com/apertus-open-source-cinema/nctrl/blob/master/Readme.md).

### webui
The webui provides several interfaces
- it uses the control daemon to then provide a basic register explorer, that allows for setting and reading registers remotely
- provides basic system information, like ip address and free memory
It will be extended to provide a end user ready interface for camera related tasks like setting exposure and gain, starting and stopping the sequencer and the hdmi feed, etc.

To learn more about the webui refer to its [readme](https://github.com/apertus-open-source-cinema/webui).

### TODO
- [x] add connection information to the home screen of the webui
- [x] update nctrl, for better lua support
- [x] move the repositories to the apertus-open-source-cinema organization
- [x] build end user ready dashboard in the webui
- [x] lua scripts in nctrl
- [x] init scripts in nctrl to load bitstream (and other init stuff, like power init and such?)
- [x] testing on micro
- [x] testing on beta

### open questions
- ~~should we move the repositories of nctrl, the webui and the camera descriptions to the apertus-open-source-cinema organization? @sp-apertus~~

  yes, we moved them.
- should the control daemon start in "mock" mode when booting the image in qemu?
- should we wait for lua scripts in nctrl?
- should we add a second API (apart from fuse) to nctrl?

  I (@rroohhh) would argue yes, as some things are hard to do with the fuse API (like exclusive access). However as we won't port the scripts and software to the control daemon in this pull request I think we can wait with adding this new API till then. 
  Any other opinions? @anuejn : i second that

### future work
Once this pull request is done / merged, some improvement opportunities open up:
- [ ] move all the old scripts and software to the control daemon
- [ ] write a compatibility layer, so that the old scripts / command (like `cmv_reg`, `fil_reg`) still work, but now through the control daemon
- [ ] add automated testing of some components using the mock mode

### CODE REVIEW

File: software/configs/nctrl.service [hunk 1]
```diff
@@ -2,8 +2,10 @@
 Description=The AXIOM nctrl daemon to expose a fuse api
 
 [Service]
-ExecStart=/usr/axiom/bin/nctrl --mountpoint /axiom-api/ $(realpath /etc/axiom-yml)
+ExecStart=bash -c "/usr/axiom/bin/nctrl --mountpoint /axiom-api/ $(realpath /etc/axiom-yml)"
 StandardOutput=journal
+# this is due to some random crashes and not fully convincing error handling in nctrl
```
Comment (rroohhh) [inline software/configs/nctrl.service]: do you have some backtraces / a way to replicate for these crashes?

Comment (anuejn) [reply software/configs/nctrl.service]: no, sadly not :(

**Comment by anuejn** [comment]
I would vote for merging this pr soon(ish) even if not all the fancy stuff (like initialization of the camera through nctrl) works.
The software in this pr is already quite useful in its current state

**Comment by sp-apertus** [comment]
Wifi hotspot works well
website with camera webgui automatically opens, great!

HDMI livestream startup fails with the problem reported in https://github.com/apertus-open-source-cinema/axiom-beta-firmware/issues/115
/usr/axiom/script/axiom-start.sh: line 52: ./rf_sel.py: No such file or directory
/usr/axiom/script/axiom-start.sh: line 53: ./pic_jtag_pcie.py: No such file or directory

**Comment by rroohhh** [comment]
@anuejn @sp-apertus I rebase the changes on the current master. After some hardware tests I think this is good to go.

Things I would test on the beta:
- connect to the camera via hotspot and try to change something like gain using the ui

Things I would test on the mirco
- test init through nctrl

**Pull request merged**